<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1123</width>
    <height>581</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>618</width>
      <height>432</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertList">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>30</y>
      <width>91</width>
      <height>511</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="heList">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>30</y>
      <width>91</width>
      <height>511</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="faceList">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>30</y>
      <width>91</width>
      <height>511</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>0</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>0</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>0</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeX">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>470</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeY">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>500</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeZ">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>530</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeG">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>500</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeB">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>530</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="changeR">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>470</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>500</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>470</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>R</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>530</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>B</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>500</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>G</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Vert Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>450</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addVertButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>460</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Vertex</string>
    </property>
   </widget>
   <widget class="QPushButton" name="triangulateButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>490</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivideButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>520</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QPushButton" name="extrudeButton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>460</y>
      <width>121</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Extrude Face</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadOBJButton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>520</y>
      <width>231</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Load OBJ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadSkeleton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>490</y>
      <width>231</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Load Skeleton</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>20</y>
      <width>171</width>
      <height>431</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>Bones</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="rotJtXButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>460</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Joint on X-Axis</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotJtYButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>490</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Joint on Y-Axis</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rotJtZButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>520</y>
      <width>171</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Joint on Z-Axis</string>
    </property>
   </widget>
   <widget class="QPushButton" name="skinMeshButton">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>460</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Skin Mesh</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>450</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Joint Position</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jtPosZ">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>530</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>500</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>470</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jtPosY">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>500</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>530</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jtPosX">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>470</y>
      <width>68</width>
      <height>24</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1123</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
